opentelemetry/proto/profiles/v1development/eventprofile.proto

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package opentelemetry.proto.profiles.v1development;
+
+import "opentelemetry/proto/common/v1/common.proto";
+
+// Profile holds a collection of events that represent a profile.
+message Profile {
+  // Data
+  repeated Event events = 1;
+
+  // Dictionary Tables
+  repeated EventType event_types = 3; // first element is the default event type
+  repeated Stack stacks = 4; // first element should be an empty stack (no frames)
+  repeated Location locations = 5;
+  repeated Function functions = 6;
+  repeated Mapping mappings = 7;
+  repeated KeyValue attributes = 8;
+  repeated Link links = 10;
+  repeated string strings = 11; // first element SHOULD be ""
+  repeated Clock clocks = 12;
+  reserved 13 to 15; // keep some 1-byte encodable field numbers for future use
+
+  // Meta Data
+  fixed64 time_unix_nano = 16; // The end time of the profile in nanoseconds since the Unix epoch.
+  int64 duration_nanos = 17; // The duration of the profile in nanoseconds.
+  repeated uint32 comment_string_index = 18;
+  // TODO: missing period
+}
+
+// Event holds a single event in a profile. It comes in two flavors:
+//
+// 1. Aggregate Event: The count is greater than 0.
+// - time and duration are optional. Their default value is
+//   represented by by Profile.time_unix_nano and Profile.duration_nanos.
+//
+// 1. Individual Event: The count value is 0 (omitted on the wire).
+// - The time field must be used and the EventType must have a Clock associated
+//   with it.
+//
+// TODO: The above is probably a bit too simplistic. There are 32 combinations
+// for the fields time, duration, value, count and sum when considering that
+// they can be unset (0) or holding a value (>1). We need to be clear which
+// combinations have use cases and which should be considered invalid.
+message Event {
+  // The index of EventType associated with this event. The most frequent event
+  // type should use index 0 so it can be omitted on the wire.
+  uint32 event_type_index = 1;
+
+  // The index of the Stack associated with this event.
+  uint32 stack_index = 2;
```

Comment:
It would be really nice in addition to the stack if we could also include raw bytes of the event, along with some way to define the format of the Event (maybe in EventType). This goes back to the issue you and I discussed about store JFR / DotNet like events in this format someway.

Reply:
@beaubelgrave that's worth considering. protobuf's stdlib already has an [Any type](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/any.proto) that can be used to embed raw protobuf messages. Maybe we could do something similar here?

Reply:
Yeah, that Any type would work well, since it gives us a type url we can use to indicate format like we were kind of circling around in the issue discussion. I'd just want to ensure that we can put the payload in this Event object and also have an optional format specified for the payload in the EventType object.

The scenario is we dump raw bytes in the Event objects and label them as "perf_event" or "tracepoint" when we get our data from Linux tracepoints. Then we'd have the tracepoint format data within the EventType struct. That way we could have small payloads in the Event with a larger format description in the EventType that has a 1:N relationship with Event (if I read the patches right).